products/views.py

```diff
@@ -1,3 +1,38 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from .models      import Product, ProductImage, ProductInfo, ImageType,SubCategory
+
+class ProductDetail(View):
+    def get(self, request, product_id):
+        try:
+            if not Product.objects.filter(id=product_id).exists():
+                return JsonResponse({'MESSAGE' : 'PRODUCT_DOES_NOT_EXIST'}, status=404)
+
+            product = Product.objects.get(id=product_id)
+
+            images = product.productimage_set.all()
+            result={
+                'korean_name'  : product.korean_name,
+                'english_name' : product.english_name,
+                'price'        : product.price,
+                'description'  : product.description,
+                'image'        :[{
+                    'image_url'  : image.image_url,
+                    'image_type' : image.image_type.type
+                    }for image in images],
```

Comment:
```suggestion
                    }for image in product.productimage_set.all()],
```
자주 쓰이지 않고, 가독성을 해치지 않기 때문에 따로 변수에 담지 않아도 좋을 것 같습니다.

Reply:
네 수정했습니다! 감사합니다